src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -15,7 +17,72 @@ public void setup() {
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void testGetProduct() {
+        Currency usd = Currency.getInstance("USD");
+        Long id = 1L;
+        Product product = new Product(id, "test", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+        productDao.save(product);
+        try {
+            assertEquals(product, productDao.getProduct(id));
+        } catch (ProductNotFoundException e) {
+            fail("Expected product");
+        }
```

Comment:
try-catch here is unnecessary, in my opinion. In case of exception the test will anyway be failed. And this try-catch add complexity for understanding of what's tested